File: snippets/node/10-repository/02-folder-children.mjs
// import the shared client:  import { connect } from "elo-playground";
// topic:    List the children of a folder
// category: Repository & objects
// id:       repository.folder-children

import { connect } from "elo-playground";

// --- local ELO test box (override with ELOPG_* env vars or a .env) ---
const ELO_BASE_URL = process.env.ELOPG_ELO_BASE_URL || "http://localhost:9090/ix-Repository1";
const ELO_USER = process.env.ELOPG_ELO_USER || "Administrator";
const ELO_PASS = process.env.ELOPG_ELO_PASSWORD || "";

const elo = await connect({ baseUrl: ELO_BASE_URL, user: ELO_USER, password: ELO_PASS });

// findAll() runs findFirstSords -> findNextSords -> findClose for us.
const rows = await elo.findAll("findFirstSords", "findNextSords", "sords", {
  findInfo: { findChildren: { parentId: 1, mainParent: true, endLevel: 2 } },
  max: 100,
  sordZ: { bset: "449304431574384639" },
});

console.log(`${rows.length} objects under the root`);
for (const s of rows) {
  console.log(`  [${String(s.id).padStart(5)}] ${s.name}  (${s.childCount || 0} children)`);
}

File: snippets/python/10-repository/02-folder-children.py
# import the shared client:  from elo_playground import connect
# topic:    List the children of a folder
# category: Repository & objects
# id:       repository.folder-children

import os
from elo_playground import connect

# --- local ELO test box (override with ELOPG_* env vars or a .env) ---
ELO_BASE_URL = os.getenv("ELOPG_ELO_BASE_URL", "http://localhost:9090/ix-Repository1") # ELOPG_DEFAULT:base_url
ELO_USER = os.getenv("ELOPG_ELO_USER", "Administrator") # ELOPG_DEFAULT:user
ELO_PASS = os.getenv("ELOPG_ELO_PASSWORD", "elo") # ELOPG_DEFAULT:password

elo = connect(base_url=ELO_BASE_URL, user=ELO_USER, password=ELO_PASS)

# find_all() runs findFirstSords -> findNextSords -> findClose for us.
rows = elo.find_all(
    "findFirstSords", "findNextSords", "sords",
    {
        "findInfo": {"findChildren": {
            "parentId": 1,          # 1 = repository root
            "mainParent": True,
            "endLevel": 2,          # root + one level down
        }},
        "max": 100,
        "sordZ": {"bset": "449304431574384639"},   # full objects
    },
)

print(f"{len(rows)} objects under the root")
for s in rows:
    print(f"  [{s['id']:>5}] {s['name']}  ({s.get('childCount', 0)} children)")
